src/core/lib/resource_quota/arena.cc

```diff
@@ -58,24 +65,25 @@ Arena::~Arena() {
     gpr_free_aligned(z);
     z = prev_z;
   }
-#ifdef GRPC_ARENA_TRACE_POOLED_ALLOCATIONS
-  gpr_log(GPR_ERROR, "DESTRUCT_ARENA %p", this);
-#endif
 }
 
 RefCountedPtr<Arena> Arena::Create(size_t initial_size,
                                    RefCountedPtr<ArenaFactory> arena_factory) {
-  return RefCountedPtr<Arena>(new (ArenaStorage(initial_size)) Arena(
-      initial_size, 0, std::move(arena_factory)));
+  void* p = ArenaStorage(initial_size);
+  return RefCountedPtr<Arena>(
+      new (p) Arena(initial_size, std::move(arena_factory)));
 }
 
-Arena::Arena(size_t initial_size, size_t initial_alloc,
-             RefCountedPtr<ArenaFactory> arena_factory)
-    : total_used_(GPR_ROUND_UP_TO_ALIGNMENT_SIZE(initial_alloc)),
-      initial_zone_size_(initial_size),
+Arena::Arena(size_t initial_size, RefCountedPtr<ArenaFactory> arena_factory)
+    : initial_zone_size_(initial_size),
+      total_used_(arena_detail::BaseArenaContextTraits::ContextSize()),
       arena_factory_(std::move(arena_factory)) {
-  arena_factory_->allocator().Reserve(
-      GPR_ROUND_UP_TO_ALIGNMENT_SIZE(initial_alloc));
+  for (size_t i = 0; i < arena_detail::BaseArenaContextTraits::NumContexts();
```

Comment:
Would this be more efficient as just a `memset()`?

Reply:
it should be reduced to that anyway